### PR TITLE
New package: bottles-2021.11.28

### PR DESCRIPTION
**Pull request by andry-dev**
Bottles is an application that manages wine prefixes, like lutris but more broad in scope.
~~As of March 2, 2021, bottles is in **alpha** so it shouldn't be merged now. This PR exists to keep track of it and allow a merge whenever it's stable enough. I'll keep the PR updated and test it as soon as I'm able to.~~
**UPDATE December 2, 2021**: It's now fine to merge Bottles: it has been stable for a few months now. Sorry for the wait regarding this PR but I've been extremely busy with real life.

Repository: https://github.com/bottlesdevs/Bottles
Site: https://usebottles.com/

### CODE REVIEW

File: srcpkgs/bottles/template [hunk 1]
```diff
@@ -0,0 +1,14 @@
+# Template file for 'bottles'
+pkgname=bottles
+version=2.0.5
+revision=1
+wrksrc=Bottles-${version}
+build_style=meson
+hostmakedepends="gettext pkg-config glib-devel desktop-file-utils appstream-glib"
```
Comment (not-chicken) [inline srcpkgs/bottles/template:7]: I think `desktop-file-utils` and `appstream-glib` can be moved to `checkdepends`. They are only required to run tests from what I see [here](https://github.com/bottlesdevs/Bottles/blob/master/data/meson.build).

**Comment by francescomasala** [comment]
By chance you can make the package available only for x86_64? This because wine runs only on that architecture.

**Comment by not-chicken** [comment]
> By chance you can make the package available only for x86_64? This because wine runs only on that architecture.

That's generally discouraged. See https://github.com/void-linux/void-packages/pull/22588#issuecomment-639074731 for explanation.

**Comment by francescomasala** [comment]
> > By chance you can make the package available only for x86_64? This because wine runs only on that architecture.
> 
> That's generally discouraged. See https://github.com/void-linux/void-packages/pull/22588#issuecomment-639074731 for explanation.

Thanks!

File: srcpkgs/bottles/template (hunk 1 above)
Comment (andry-dev) [reply srcpkgs/bottles/template:7]: Yeah they can be moved, the package still compiles fine.

The checking phase fails but that's a really minor problem with upstream related to the screenshots in the repository, so it's not a showstopper.

**Comment by Meister1593** [comment]
when it will be merged?

**Comment by ericonr** [comment]
Needs to be updated, but template currently looks ok.

**Comment by andry-dev** [comment]
@ericonr Hi, I'm back! Sorry for the wait.

I've _finally_ updated the PR. I had to patch some checks out for `patool` because otherwise it would fail the tests.

**Comment by Anachron** [comment]
Misses `python3-Markdown` in `depends`:

``sh`
$ bottles
Traceback (most recent call last):
  File "/bin/bottles", line 56, in <module>
    from bottles import main
  File "/usr/share/bottles/bottles/main.py", line 33, in <module>
    from .window import MainWindow
  File "/usr/share/bottles/bottles/window.py", line 26, in <module>
    from .backend.manager import Manager
  File "/usr/share/bottles/bottles/backend/manager.py", line 40, in <module>
    from .installer import InstallerManager
  File "/usr/share/bottles/bottles/backend/installer.py", line 20, in <module>
    import markdown
ModuleNotFoundError: No module named 'markdown'
```

After rebuilding, `bottles` now launch:
```
$ bottles
2022-01-10 11:08:29 ERROR: Data file not found. Creating new one.
2022-01-10 11:08:29 INFO: Runners path doesn't exist, creating now.
2022-01-10 11:08:29 INFO: Bottles path doesn't exist, creating now.
2022-01-10 11:08:29 INFO: Dxvk path doesn't exist, creating now.
2022-01-10 11:08:29 INFO: Vkd3d path doesn't exist, creating now.
2022-01-10 11:08:29 INFO: Nvapi path doesn't exist, creating now.
2022-01-10 11:08:29 INFO: Temp path doesn't exist, creating now.
error: unable to find wine executable.  this shouldn't happen.
2022-01-10 11:08:29 INFO: Runners found: [sys-]
2022-01-10 11:08:30 INFO: Bottles gestartet!
```

**Comment by eth0OS** [comment]
Here I still get the error:

> Traceback (most recent call last):
>   File "/bin/bottles", line 56, in <module>
>     from bottles import main
>   File "/usr/share/bottles/bottles/main.py", line 33, in <module>
>     from .window import MainWindow
>   File "/usr/share/bottles/bottles/window.py", line 26, in <module>
>     from .backend.manager import Manager
>   File "/usr/share/bottles/bottles/backend/manager.py", line 41, in <module>
>     from .dependency import DependencyManager
>   File "/usr/share/bottles/bottles/backend/dependency.py", line 21, in <module>
>     import patoolib
> ModuleNotFoundError: No module named 'patoolib'

Since I believe there is no python3-patool in void's repo.

**Comment by Anachron** [comment]
Its included in this PR... how did you install this pkg without installing patoolib?

**Comment by Anachron** [comment]
How did you compile & install your `bottles`? This PR already has `patool` included.

**Comment by eth0OS** [comment]
> How did you compile & install your `bottles`? This PR already has `patool` included.

At first I I pulled void-packages, as told in the documentation.
Then I copied the PR's template into the void-packages/srcpkgs/bottles/template file and ran `./xbps-src pkg bottles`:
This procedure returned me this error
```
=> xbps-src: updating repositories for host (x86_64)...
[*] Updating repository 'https://alpha.de.repo.voidlinux.org/current/x86_64-repodata' ...
x86_64-repodata: 1713KB [avg rate: 3493KB/s]
[*] Updating repository 'https://alpha.de.repo.voidlinux.org/current/nonfree/x86_64-repodata' ...
[*] Updating repository 'https://alpha.de.repo.voidlinux.org/current/debug/x86_64-repodata' ...
x86_64-repodata: 773KB [avg rate: 49GB/s]
[*] Updating repository 'https://alpha.de.repo.voidlinux.org/current/multilib/x86_64-repodata' ...
x86_64-repodata: 663KB [avg rate: 23GB/s]
[*] Updating repository 'https://alpha.de.repo.voidlinux.org/current/multilib/nonfree/x86_64-repodata' ...
=> xbps-src: updating software in / masterdir...
=> xbps-src: cleaning up / masterdir...
=> bottles-2021.11.28_1: removing autodeps, please wait...
=> bottles-2021.11.28_1: building [meson] for x86_64...
   [host] gettext-0.21_4: found (https://alpha.de.repo.voidlinux.org/current)
   [host] pkg-config-0.29.2_3: found (https://alpha.de.repo.voidlinux.org/current)
   [host] glib-devel-2.70.2_1: found (https://alpha.de.repo.voidlinux.org/current)
   [host] meson-0.60.3_1: found (https://alpha.de.repo.voidlinux.org/current)
   [target] python3-devel-3.10.1_1: found (https://alpha.de.repo.voidlinux.org/current)
   [target] libhandy1-devel-1.5.0_1: found (https://alpha.de.repo.voidlinux.org/current)
=> ERROR: bottles-2021.11.28_1: target dependency 'python3-patool' does not exist!
```

Later, I tried removing the pytho3-patool from the template, the compilation and building finished succesfully but of course the application didn't run without the patool module.

Did I miss anything?

**Comment by Anachron** [comment]
Even without the python3-Markdown dependency? If so, do you already have it installed by another pkg?

**Comment by eth0OS** [comment]
Bottles won't run without python3-Markdown, it will output this instead:

> Traceback (most recent call last):
>   File "/bin/bottles", line 56, in <module>
>     from bottles import main
>   File "/usr/share/bottles/bottles/main.py", line 33, in <module>
>     from .window import MainWindow
>   File "/usr/share/bottles/bottles/window.py", line 26, in <module>
>     from .backend.manager import Manager
>   File "/usr/share/bottles/bottles/backend/manager.py", line 40, in <module>
>     from .installer import InstallerManager
>   File "/usr/share/bottles/bottles/backend/installer.py", line 20, in <module>
>     import markdown
> ModuleNotFoundError: No module named 'markdown'

**Comment by Anachron** [comment]
Did you read my message from January 10th?

**Comment by eth0OS** [comment]
> Did you read my message from January 10th?

Yes, I did. I missed the patool template that resides inside this PR. Compiling it and installing patool with the provided template allowed me to compile, install and run bottles fine.

**Comment by andry-dev** [comment]
@Anachron 

Sorry for the extreme delay, but unfortunately life got hectic in the last few months and I'm afraid I don't have the time to maintain bottles anymore. I personally know @francescomasala and the huge work the team put on the project, so I think I can no longer do the package any justice keeping up with the updates and possible new upstream dependencies. I hope someone else can continue the PR from my templates.

Again, sorry for the inconvenience.

**Pull request closed without merging**